extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMap.kt

```diff
@@ -0,0 +1,94 @@
+package com.mapbox.maps.extension.androidauto
+
+import android.graphics.Rect
+import androidx.car.app.CarContext
+import androidx.lifecycle.Lifecycle
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.MapInitOptions
+import com.mapbox.maps.MapboxExperimental
+
+/**
+ * This is the main entry point for controlling the Mapbox car map surface.
+ *
+ * You can add the [MapboxCarMap] to your entire [androidx.car.app.Session] using the [Lifecycle].
+ * And then any [androidx.car.app.Screen] that is using a supported template will automatically
+ * show the Mapbox map. You can also, specify a [MapboxCarMap] for each individual
+ * [androidx.car.app.Screen] by constructing the [MapboxCarMap] with the screen Lifecycle.
+ *
+ * Supported templates include:
+ *  - [androidx.car.app.navigation.model.NavigationTemplate]
+ *  - [androidx.car.app.navigation.model.RoutePreviewNavigationTemplate]
+ *  - [androidx.car.app.navigation.model.PlaceListNavigationTemplate]
+ *
+ * Customize your [MapboxCarMap] with your own implementations of [MapboxCarMapObserver].CarMapSurfaceOwnerTest
+ * Use the [registerObserver] and [unregisterObserver] functions to load and detach the observers.
+ *
+ * @param mapInitOptions to initialize the [MapboxCarMapSurface]
+ */
+@MapboxExperimental
+class MapboxCarMap(
+  private val mapInitOptions: MapInitOptions
+) {
+  /**
+   * Accessor for the carContext provided to the MapInitOptions. This makes it easier to create
+   * screens with the MapboxCarMap in the constructor.
+   *
+   * For example:
+   *  class YourMapScreen(val mapboxCarMap: MapboxCarMap) : Screen(mapboxCarMap.carContext) {
+   *
+   * The carContext can also be found in the [MapboxCarMapObserver] callbacks. Make sure to
+   * call [MapboxCarMap.clearObservers] when your car session is destroyed.
+   */
+  val carContext: CarContext = mapInitOptions.context as? CarContext
+    ?: throw IllegalArgumentException("You must construct a MapboxCarMap with the CarContext")
+
+  private val carMapSurfaceOwner = CarMapSurfaceOwner()
+  private val carMapSurfaceCallback = CarMapSurfaceCallback(
+    carContext,
+    carMapSurfaceOwner,
+    mapInitOptions
+  )
+
+  init {
+    carMapSurfaceCallback.onCreate()
```

Comment:
Since it's not called from callback - any reasons to not make this `onCreate` a constructor of `CarMapSurfaceCallback` instead?

Reply:
Discussed some internally, preferring the immutable constructor. resolving this thread

```
MapboxCarMap(MapInitOptions(carContext))
```